doc/bfast/fixed_angle_broadband_simulations_in_Meep.md

```diff
@@ -0,0 +1,154 @@
+(8) can be written as
+```math
+\vec{K} = -i\omega(1+\frac{i\sigma_{D}}{\omega})\vec{C} - i\omega\vec{F}. \tag{9}
+```
+This new field satisfies the equation:
+```math
+\vec{F} = \vec{\bar{k}}\times\vec{H}, \tag{10}
+```
+where
+```math
+\vec{\bar{k}} =\frac{1}{\omega}\begin{pmatrix} k_{x} \\ k_{y}\\0 \end{pmatrix} \ = n \begin{pmatrix} \sin{\theta}\cos{\phi} \\ \sin{\theta}\sin{\phi}\\0 \end{pmatrix} \ \tag{11}
+```
+and so $\vec{\bar{k}}$ is the wave vector with its frequency dependence
+removed. $\theta$ and $\phi$ are the propagating direction angles, $n$ is the refractive index of the source medium and c,
+the speed of light is taken to be 1. Therefore by defining $\vec{F}$,
+the angle of the incident wave is fixed. Equation
+(10) can be discretized as:
+```math
+\vec{F}^{n+1}=2\bar{\vec{k}}\times\vec{H}^{n+0.5} -\vec{F}^{n} . \tag{12}
+```
+Transforming equation (9)
+to the time domain gives:
+```math
+\vec{K} = \frac{\partial \vec{C}}{\partial t}+\sigma_{D}\vec{C}+\frac{\partial \vec{F}}{\partial t} . \tag{13}
+```
+This can be discretized as:
+```math
+\vec{K}^{n+0.5}=\frac{\vec{C}^{n+1}-\vec{C}^n}{\Delta t}+\sigma_{D}\frac{\vec{C}^{n+1}+\vec{C}^n}{2} + \frac{\vec{F}^{n+1}-\vec{F}^{n}}{\Delta t} \tag{14}
+```
+and then solved to update the value of $\vec{C}$ using:
+```math
+\vec{C}^{n+1}=(1+\frac{\sigma_{D} \Delta t}{2})^{-1} [(1-\frac{\sigma_D \Delta t}{2}) \vec{C}^n+\Delta t\vec{K}^{n+0.5}+\vec{F}^{n}-\vec{F}^{n+1}] . \tag{15}
+```
+All other equations are unaffected by these changes.
+
+A new field must be introduced because $\vec{H}$ is defined at
+$n+\frac{1}{2}$ timesteps whereas $\vec{C}$ is defined at $n$ timesteps,
+where $n$ is an integer. As a result, if the derivative in $\vec{F}$ in
+equation (14) was replaced with
+```math
+\vec{\bar{k}}\times(\frac{\vec{H}^{n+0.5}-\vec{H}^{n-0.5}}{\Delta t}), \tag{16}
+```
+only first order accuracy would be achieved, since this is a backward
+difference scheme. To achieve second order accuracy would require
+$\vec{H}^{n+1.5}$ to be known.
+
+Application to the source code
+=========
+
+$\vec{C}$ is first time-stepped using the original function in the MEEP source code, i.e. using
+```math
+\vec{C}^{n+1}=(1+\frac{\sigma_{D} \Delta t}{2})^{-1} [(1-\frac{\sigma_D \Delta t}{2}) \vec{C}^n+\Delta t\vec{K}^{n+0.5}]. \tag{17}
+```
+When fixed angle broadband simulations are required, a new function is called which recovers equation (15) using
+```math
+\vec{C'}^{n+1} = (1+\frac{\sigma_{D} \Delta t}{2})^{-1} (\vec{F}^{n}-\vec{F}^{n+1}) + \vec{C}^{n+1}. \tag{18}
+```
+Similarly, all other fields which depend on the value of $\vec{C}$ have new terms added to them, multiplied by the relevant conductivity. Therefore the original functions in MEEP remain unchanged.
+
+Stability
+=========
+
+As the incident angle increases, the maximum possible $\Delta t$ value
+decreases, following the formula:
+```math
+\frac{c\Delta t}{\Delta x} \leq \frac{(1-\sin(\theta))}{\sqrt{D}} \tag{17}
+```
```

Comment:
It's a little surprising to me that this depends on sin(θ) and not sin(θ)^2 — I would have thought that the stability condition would be the same for θ and –θ.

Let's double-check this formula against the bfast literature?

Reply:
Nope, that is exactly the formula in the paper:

![image](https://github.com/NanoComp/meep/assets/2913679/2d808270-bac3-41c7-8a37-169f20b79db1)